wallet/wallet/hd_wallet.go

```diff
@@ -268,6 +268,15 @@ func (w *HDWallet) VerifyAny(pubKey string, data, sig []byte) (bool, error) {
 	return keyPair.VerifyAny(data, sig)
 }
 
+func (w *HDWallet) CheckTx(pubKey string, data []byte) (*Signature, error) {
+	keyPair, ok := w.keyRing.FindPair(pubKey)
+	if !ok {
+		return nil, ErrPubKeyDoesNotExist
+	}
+
+	return keyPair.Sign(data)
+}
+
```

Comment:
That part is redundant with the `HDWAllet.SignTx()` method. It's fine to use the `SignTx()` method.